src/main/java/org/nypl/harvester/sierra/processor/StreamPoster.java

```diff
@@ -46,8 +47,8 @@ private void sendToKinesis(List<Item> items) throws SierraHarvesterException{
 							
 							@Override
 							public void process(Exchange kinesisRequest) throws Exception {
-								kinesisRequest.getIn().setHeader(HarvesterConstants.KINESIS_PARTITION_KEY, 
-										Integer.parseInt(System.getenv("KinesisPartitionKey")));
+								kinesisRequest.getIn().setHeader(HarvesterConstants.KINESIS_PARTITION_KEY,
+										UUID.randomUUID());
```

Comment:
Data records with the same partition key goes to the same shard for distributing data into multiple shards make sense. However, lets talk about how do we change shard count dynamically?

cc - @beefoo @EdwinGuzman